lib/std/private/strimpl.nim

```diff
@@ -51,3 +51,19 @@ template cmpIgnoreCaseImpl*(a, b: typed, firstCharCaseSensitive: static bool = f
     if result != 0: return
     inc i
   result = aLen - bLen
+
+template startsWithImpl*(s, prefix: typed) =
+  # s, prefix are string or cstring
+  var i = 0
+  while true:
+    if i >= prefix.len: return true
```

Comment:
once https://github.com/nim-lang/Nim/issues/16405 is fixed, prefix.len will be O(n), so best to cache this value

Reply:
ok